### PR TITLE
Issue #2677664: Added configurable phone number verification path

### PR DESCRIPTION
**Pull request by dpi**
https://www.drupal.org/node/2677664
- Added new page config.
- Refactored SmsSettingsForm
- Sms settings form no longer redirects.
- Added Drupal message on setttings submission.

### CODE REVIEW

File: src/Form/SmsSettingsForm.php [hunk 1]
```diff
@@ -17,6 +21,47 @@
...
+  /**
+   * The request context.
+   *
+   * @var \Drupal\Core\Routing\RequestContext
+   */
+  protected $requestContext;
+
+  /**
+   * Constructs a SiteInformationForm object.
+   *
```
Comment (almaudoh) [inline src/Form/SmsSettingsForm.php]: ????

File: src/Form/SmsSettingsForm.php [hunk 2]
```diff
@@ -17,6 +21,47 @@
...
+
+  /**
+   * The request context.
+   *
+   * @var \Drupal\Core\Routing\RequestContext
+   */
+  protected $requestContext;
+
+  /**
+   * Constructs a SiteInformationForm object.
```
Comment (almaudoh) [inline src/Form/SmsSettingsForm.php]: ????

Comment (dpi) [inline src/Form/SmsSettingsForm.php]: Nice catch

File: src/Routing/RouteSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        [
+          '_form' => '\Drupal\sms\Form\VerifyPhoneNumberForm',
+          '_title' => 'Verify a phone number',
+        ],
+        [
+          '_permission' => 'sms verify phone number',
+        ]
+      ));
+    }
+
```
Comment (almaudoh) [inline src/Routing/RouteSubscriber.php:73]: Is it safe to fail silently in the case the `$path_verify` is malformed?

Comment (dpi) [inline src/Routing/RouteSubscriber.php:73]: its a just-in-case, such as if there was a problem with installing module config (happens sometimes if install fails midway), or a config import/export problem.

File: src/Tests/SmsFrameworkSettingsTest.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    ]);
+    $this->drupalLogin($account);
+
+    // Test invalid path.
+    $edit = [
+      'default_gateway' => 'log',
+      'pages[verify]' => $this->randomMachineName() . '/' . $this->randomMachineName(),
+    ];
+    $this->drupalPostForm(Url::fromRoute('sms.settings'), $edit, 'Save configuration');
+    $this->assertRaw(t("Path must begin with a '/' character."));
```
Comment (almaudoh) [inline src/Tests/SmsFrameworkSettingsTest.php:34]: We shouldn't be using t() in asserts

File: src/Tests/SmsFrameworkSettingsTest.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    // Test submission
+    $edit = [
+      'default_gateway' => 'log',
+      'pages[verify]' => '/' . $this->randomMachineName(),
+    ];
+    $this->drupalPostForm(Url::fromRoute('sms.settings'), $edit, 'Save configuration');
+    $this->assertRaw(t('SMS settings saved.'));
```
Comment (almaudoh) [inline src/Tests/SmsFrameworkSettingsTest.php:42]: t() in asserts...

File: src/Tests/SmsFrameworkSettingsTest.php (hunk 4 above)
Comment (dpi) [inline src/Tests/SmsFrameworkSettingsTest.php:34]: I forget the whole reason why, but its something to do with the parent site leaking language. Using t in asserts is acceptable, even if not using placeholders. But if you render with t, you can (should?) use it in tests.

Search "assertRaw(t(" or assertText(t(" for plenty of instances.

**Comment by dpi** [comment]
hmm these random test failures are interesting

**Pull request closed without merging**